go/database/mpt/proof.go

```diff
@@ -225,6 +227,26 @@ func (p *proofExtractionVisitor) Visit(node Node, info NodeInfo) VisitResponse {
 		return VisitResponseAbort
 	}
 
+	if p.updateChildHashes {
+		switch n := node.(type) {
+		case *ExtensionNode:
+			if err := updateChildrenHashesComputeEmbedded(p.nodeSource, n, []NodeReference{n.next}); err != nil {
+				p.err = err
+				return VisitResponseAbort
+			}
+		case *BranchNode:
+			if err := updateChildrenHashesComputeEmbedded(p.nodeSource, n, n.children[:]); err != nil {
+				p.err = err
+				return VisitResponseAbort
+			}
+		case *AccountNode:
+			if err := updateChildrenHashesComputeEmbedded(p.nodeSource, n, []NodeReference{n.storage}); err != nil {
+				p.err = err
+				return VisitResponseAbort
+			}
+		}
```

Comment:
I would prefer that the function `updateChildrenHashesComputeEmbedded` would handle the extraction of the list of children instead of the caller and then pass it as an argument.

Reply:
ok, code inlined into this switch